Runtime/Scripts/PersistentConnection.cs

```diff
@@ -296,104 +327,176 @@ IEnumerator OnIntervalEdgeEventsLatency(EdgeEventsConnection connection, string
       {
         connection.TestPingAndPostLatencyUpdate(host, location).ConfigureAwait(false);
       }
+      Logger.Log("EdgeEvents Posting latency update, Location to send [" + location.Latitude + ", " + location.Longitude + "]");
       latencyUpdatesCounter++;
       yield return StartCoroutine(OnIntervalEdgeEventsLatency(connection, host));
     }
 
-    async void HandleReceivedEvents(ServerEdgeEvent edgeEvent)
+    void HandleReceivedEvents(ServerEdgeEvent edgeEvent)
     {
       Logger.Log("Received event type: " + edgeEvent.EventType);
-      FindCloudletEventTrigger trigger = event_trigger_dict[edgeEvent.EventType];
-      if (config.newFindCloudletEventTriggers.Contains(trigger))
+      List<FindCloudletEventTrigger> fcTriggers = config.newFindCloudletEventTriggers;
+      switch (edgeEvent.EventType)
       {
-        switch (trigger)
-        {
-          case FindCloudletEventTrigger.Error:
-            PropagateError(FindCloudletEventTrigger.Error, edgeEvent.ErrorMsg);
-            return;
-          case FindCloudletEventTrigger.LatencyTooHigh:
-            if (edgeEvent.EventType == ServerEventType.EventLatencyRequest)
+        case ServerEventType.EventInitConnection:
+          Logger.Log("Successfully initiated persistent edge event connection");
+          return;
+        case ServerEventType.EventLatencyRequest:
+          if (fcTriggers.Contains(FindCloudletEventTrigger.LatencyTooHigh))
+          {
+            StartCoroutine(RespondToLatencyRequest());
+          }
+          return;
+        case ServerEventType.EventLatencyProcessed:
+          if (fcTriggers.Contains(FindCloudletEventTrigger.LatencyTooHigh))
+          {
+            latestServerStats = edgeEvent.Statistics;
+            if(!latencyProcessingInProgress)
             {
-              await RespondToLatencyRequest();
+              startLatencyProcessing = true;
+            }
+          }
+          return;
+        case ServerEventType.EventAppinstHealth:
+          if (fcTriggers.Contains(FindCloudletEventTrigger.AppInstHealthChanged))
+          {
+            if (edgeEvent.NewCloudlet == null)
```

Comment:
sorry i wasn't clear. but all these checks should also have a check to see if edgeEvent.ErrorMsg != "" (ie. `if (edgeEvent.NewCloudlet == null || edgeEvent.ErrorMsg != "")`). Then you can add that errormsg to your ProgagateError message